### DISCUSSION TITLE
Pregunta complejidades

### DISCUSSION THREAD

**Discussion by GuilleStankeS** [IIC2133-PUC/2024-2#82] [Tarea 1] [answered]
Hola, tengo una duda respecto a las complejidades que se piden... es normal que en algunos casos se pueda hacer de forma que en el peor caso la complejidad sea menor a la complejidad solicitada (por ejemplo: evento ENTER de parte 1, obtener una complejidad de O(log(n)) en ves de O(log(n) + log(s))), o se solicita esa complejidad por alguna razón en específico? (como que se debería hacer de cierta forma que el peor caso sea de esa complejidad y sirva también como modo de "hint" de como debemos trabajar la complejidad que piden)

**Comment by Segismundios** [comment, marked as answer]
Hola @GuilleStanke. No hay problema con que logres resolver un evento con una complejidad menor a la exigida, lo importante es que no sea mayor. Sí es cierto que la complejidad solicitada puede servir como hint a una de las posibles soluciones para el problema planteado. Te recomendaría pensar de dónde sale la parte de log(s). Si aún ves que cumples con todo lo solicitado en log(n), entonces está bien. 
Saludos!
